### PR TITLE
[Security] renamed memory provider and a tip about encoder for all user types

### PR DESCRIPTION
**Pull request by HeahDude**
A follow up of https://github.com/symfony/recipes/pull/741.

### CODE REVIEW

File: security.rst [hunk 1]
```diff
@@ -421,20 +423,22 @@ To fix this, add an ``encoders`` key:
...
 
-User providers load user information and put it into a ``User`` object. If
-you :doc:`load users from the database </security/entity_provider>`
+User providers load user information and put it into a :class:`Symfony\Component\Security\Core\User\UserInterface`
+implementation. If you :doc:`load users from the database </security/entity_provider>`
 or :doc:`some other source </security/custom_provider>`, you'll
-use your own custom User class. But when you use the "in memory" provider,
-it gives you a ``Symfony\Component\Security\Core\User\User`` object.
+use your own custom User class. But when you use the "in memory" provider type,
+it gives you a :class:`Symfony\Component\Security\Core\User\User` object.
```
Comment (javiereguiluz) [inline security.rst]: Minor issue here: when using `:class:` you must double all slashes.

Comment (HeahDude) [reply security.rst]: Fixed, thanks!

File: security.rst [hunk 2]
```diff
@@ -449,6 +465,67 @@ you who you are and what roles you have:
...
+    .. configuration-block::
+
+        .. code-block:: yaml
+
+            # app/config/security.yml
+            security:
+                encoders:
+                    Symfony\Component\Security\Core\User\UserInterface: bcrypt
+
+                    # would be equivalent to:
```
Comment (wouterj) [inline security.rst]: Let's use present tense: "is equivalent to:"

**Review by wouterj — APPROVED** [review]

File: security.rst (hunk 2 above)
Comment (HeahDude) [reply security.rst]: Fixed.

**Pull request merged**